lib/services/livesync/ios-device-livesync-service.ts

```diff
@@ -31,78 +32,140 @@ class IOSLiveSyncService extends DeviceLiveSyncServiceBase<Mobile.IiOSDevice> im
+			let shouldRestart = _.some(otherFiles, (localToDevicePath: Mobile.ILocalToDevicePathData) => !this.$liveSyncProvider.canExecuteFastSync(localToDevicePath.getLocalPath(), deviceAppData.platform));
+
+			if (shouldRestart) {
+				this.restartApplication(deviceAppData).wait();
+
+				return;
+			}
+
+			if (!this.$options.liveEdit && scriptFiles.length) {
+				this.restartApplication(deviceAppData).wait();
+
+				return;
+			}
+
+			if (this.setupSocketIfNeeded().wait()) {
+				this.liveEdit(scriptFiles);
+				this.reloadPage(deviceAppData, otherFiles).wait();
+			}
+		}).future<void>()();
+	}
+
+	private restartApplication(deviceAppData: Mobile.IDeviceAppData): IFuture<void> {
 		let projectData: IProjectData = this.$injector.resolve("projectData");
 		return this.device.applicationManager.restartApplication(deviceAppData.appIdentifier, projectData.projectName);
 	}
 
-	protected reloadPage(deviceAppData: Mobile.IDeviceAppData): IFuture<void> {
+	private reloadPage(deviceAppData: Mobile.IDeviceAppData, localToDevicePaths: Mobile.ILocalToDevicePathData[]): IFuture<void> {
 		return (() => {
-			let timeout = 9000;
-			if (this.device.isEmulator) {
-				if (!this.socket) {
-					helpers.connectEventually(() => net.connect(IOSLiveSyncService.BACKEND_PORT), (socket: net.Socket) => {
-						this.socket = socket;
-						this.attachEventHandlersIfNecessary();
-						this.sendPageReloadMessage();
-					});
-				} else {
-					this.sendPageReloadMessage();
-				}
-			 	this.$iOSEmulatorServices.postDarwinNotification(this.$iOSNotification.attachRequest).wait();
-			} else {
-				if(!this.socket) {
-					this.$iOSSocketRequestExecutor.executeAttachRequest(this.device, timeout).wait();
-					this.socket = this.device.connectToPort(IOSLiveSyncService.BACKEND_PORT);
-					this.attachEventHandlersIfNecessary();
-				}
-				this.sendPageReloadMessage();
+			if (localToDevicePaths.length) {
+				let message = JSON.stringify({
+					method: "Page.reload",
+					params: {
+						ignoreCache: false
+					},
+					id: ++currentPageReloadId
+				});
+
+				this.sendMessage(message);
 			}
 		}).future<void>()();
 	}
 
-	private attachEventHandlersIfNecessary(): void {
-		if(this.$options.watch) {
-			this.attachProcessExitHandlers();
-			this.attachSocketCloseEvent();
-		}
+	private liveEdit(localToDevicePaths: Mobile.ILocalToDevicePathData[]) {
+		return (() => {
+			_.each(localToDevicePaths, localToDevicePath => {
+				let content = this.$fs.readFile(localToDevicePath.getLocalPath()).wait().toString();
+				let message = JSON.stringify({
+					method: "Debugger.setScriptSource",
+					params: {
+						scriptUrl: localToDevicePath.getDevicePath(),
+						scriptSource: content
+					},
+					id: ++currentPageReloadId
+				});
+
+				this.sendMessage(message);
+			});
+		}).future<void>()();
 	}
 
-	private attachSocketCloseEvent(): void {
+	private attachEventHandlers(): void {
+		this.attachProcessExitHandlers();
```

Comment:
you can replace this with
```TypeScript
this.$processService.attachToProcessExitSignals(this, this.destroySocket);
```
This will prevent multiple handlers for `exit`, `SIGINT` and `SIGTERM` events.